tool/tsh/common/db.go

```diff
@@ -1248,115 +1292,95 @@ func isMFADatabaseAccessRequired(ctx context.Context, tc *client.TeleportClient,
 //
 // If logged into multiple databases, returns an error unless one specified
 // explicitly via --db flag.
-func pickActiveDatabase(cf *CLIConf, tc *client.TeleportClient) (*tlsca.RouteToDatabase, error) {
-	profile, err := tc.ProfileStatus()
-	if err != nil {
-		return nil, trace.Wrap(err)
+func pickActiveDatabase(cf *CLIConf, tc *client.TeleportClient, routes []tlsca.RouteToDatabase) (*tlsca.RouteToDatabase, error) {
```

Comment:
The input parameter for this function is `routes []tlsca.RouteToDatabase`   that orginates from 
```
routes, err := profile.DatabasesForCluster(tc.SiteName) 
``` 
notice that the `DatabasesForCluster`  call already returns active databases   [findActiveDatabases](https://github.com/gravitational/teleport/blob/master/lib/client/profile.go#L554)

Reply:
Because it calls getDatabaseInfo - that func will resolve the name, labels, and query selectors to a single database and then search for that database's full name in the active db routes passed to it. getDatabaseInfo doesn't necessarily find an active db though, it just looks for one so it can copy the db user/schema from an active route

Reply:
It needs to pick among the active routes it is passed. Doing so requires that it resolve the full db name.

It has to filter routes by name prefix, labels, query predicate, and then look for:
1. an exact full name match
2. an exact match of a discovered name
3. a single remaining database

Reply:
also, the reason I made these functions take the `[]tlsca.RouteToDatabase` as an argument instead of just getting them from the profile, was to make unit testing them much easier - allows me to control which databases are "logged into" in tests.